File: src/main/java/com/hedvig/backoffice/graphql/dataloaders/ClaimLoader.java
package com.hedvig.backoffice.graphql.dataloaders;

import java.util.UUID;
import com.hedvig.backoffice.graphql.types.Claim;
import org.dataloader.DataLoader;
import org.dataloader.DataLoaderOptions;
import org.springframework.stereotype.Component;

@Component
public class ClaimLoader extends DataLoader<UUID, Claim> {
  public ClaimLoader(ClaimBatchLoader claimBatchLoader) {
    super(claimBatchLoader, DataLoaderOptions.newOptions().setCachingEnabled(false));
  }
}

File: src/main/java/com/hedvig/backoffice/graphql/dataloaders/ClaimLoader.kt
package com.hedvig.backoffice.graphql.dataloaders

import com.hedvig.backoffice.graphql.types.Claim
import org.dataloader.DataLoader
import org.dataloader.DataLoaderOptions
import org.springframework.stereotype.Component
import java.util.UUID

@Component
class ClaimLoader(
    claimBatchLoader: ClaimBatchLoader
) : DataLoader<UUID, Claim>(
    claimBatchLoader,
    DataLoaderOptions.newOptions().setCachingEnabled(false)
)
